bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschHttpClient.java

```diff
@@ -0,0 +1,187 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.boschshc.internal.devices.bridge;
+
+import static org.eclipse.jetty.http.HttpMethod.GET;
+
+import java.nio.charset.StandardCharsets;
+import java.security.KeyStoreException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateEncodingException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * HTTP client using own context with private & Bosch Certs
+ * to pair and connect to the Bosch Smart Home Controller.
+ *
+ * @author Gerd Zanker - Initial contribution
+ */
+@NonNullByDefault
+public class BoschHttpClient extends HttpClient {
+    private final Logger logger = LoggerFactory.getLogger(BoschHttpClient.class);
+
+    private String ipAddress;
+    private String systempassword;
+
+    private Gson gson = new Gson();
+
+    public BoschHttpClient(String ipAddress, String systempassword, SslContextFactory sslContextFactory) {
+        super(sslContextFactory);
+        this.ipAddress = ipAddress;
+        this.systempassword = systempassword;
+    }
+
+    private String getCertFromSslContextFactory() throws KeyStoreException, CertificateEncodingException {
+        Certificate cert = this.getSslContextFactory().getKeyStore().getCertificate(BoschSslUtil.getBoschSHCId());
+        return Base64.getEncoder().encodeToString(cert.getEncoded());
+    }
+
+    public boolean isAccessPossible() {
+        try {
+            String url = this.createSmartHomeUrl("devices");
+            Request request = this.createRequest(url, GET);
+            ContentResponse contentResponse = request.send();
+            String content = contentResponse.getContentAsString();
+            logger.debug("Access check response complete: {} - return code: {}", content, contentResponse.getStatus());
+            return true;
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.debug("Access check response failed because of {}!", e.getMessage());
+            return false;
```

Comment:
I wouldn't consider this a proper handling of the InterruptedException. You need to either throw another exception or rethrow the InterruptedException.

An InterruptedException typically means that the interrupter wants the thread to stop what it is doing and shut down. Handling the InterruptedException here would cause code higher up the stack to continue running normally, which is exactly the opposite of what the interrupter wants to happen.

Reply:
I am throwing the InterruptedException now instead of catching it, so it can be handled by the callers. In general the status of the thing will change to Offline with the exception message in the status message. In my opinion this should be appropriate for those very rare cases. If not, please give me a hint how a better handling could look like :)